frontend/src/models/models.js

```diff
@@ -143,6 +146,51 @@ module.exports = app => app.component('models', {
       for (const { path } of this.schemaPaths) {
         this.shouldExport[path] = true;
       }
+      this.filteredPaths.length = 0;
+      this.filteredPaths = [...this.schemaPaths];
+      this.selectedPaths = [...this.schemaPaths];
+    },
+    addOrRemove(path) {
+      const exists = this.selectedPaths.findIndex(x => x.path == path.path);
+      if (exists > 0) { // remove
+        this.selectedPaths.splice(exists, 1);
+      } else { // add
+        this.selectedPaths.push(path);
+        this.selectedPaths = Object.keys(this.selectedPaths).sort((k1, k2) => {
+          if (k1 === '_id' && k2 !== '_id') {
+            return -1;
+          }
+          if (k1 !== '_id' && k2 === '_id') {
+            return 1;
+          }
+          return 0;
+        }).map(key => this.selectedPaths[key]);
+      }
+    },
+    filterDocuments() {
+      this.filteredPaths = Object.keys(this.selectedPaths).sort((k1, k2) => {
+        if (k1 === '_id' && k2 !== '_id') {
+          return -1;
+        }
+        if (k1 !== '_id' && k2 === '_id') {
+          return 1;
+        }
+        return 0;
+      }).map(key => this.selectedPaths[key]);
+      this.shouldShowFieldModal = false;
+      const selectedParams = this.filteredPaths.map(x => x.path).join(',');
+      console.log('what is selectedParams', selectedParams)
+
+      // Use the History API to update the URL without reloading the page
+      const newUrl = `${window.location.pathname}?fields=${selectedParams}`;
+      window.history.pushState({ path: newUrl }, '', newUrl);
```

Comment:
`pushState()` isn't really correct here because Mongoose studio uses Vue router `hashHistory`, not native browser history: https://github.com/mongoosejs/studio/blob/29dbeb7db1e4c9005ae29d29b94378b267f45fb6/frontend/src/index.js#L39. 

Use `this.$router.push()` instead with `{ path, query }` args like on this page: https://router.vuejs.org/guide/essentials/navigation.html. Should work better.